app/javascript/packages/react-hooks/use-object-memo.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from 'react';
+
+/**
+ * React hook which creates a memoized object whose reference changes when values of the object
+ * changes.
+ *
+ * This can be useful in situations like object context values, since without memoization an object
+ * context would trigger re-renders of all consumers on every update.
+ *
+ * Note that the keys of the object must remain the same for the lifecycle of the component for the
+ * hook to work correctly.
+ *
+ * @param object Object to memoize.
+ *
+ * @return Memoized object.
+ */
+const useObjectMemo = <T extends object>(object: T): T =>
+  useMemo(() => object, Object.values(object));
```

Comment:
Do we need to worry about order here? Does this assume that we're always using like ad-hoc object creation in the same spot, so the keys will remain ordered constantly?

I tried this out in node and the order of creation did seem to matter:
```node
> Object.values({ x: 1, y: 2 })
[ 1, 2 ]
> Object.values({ y: 1, x: 2})
[ 1, 2 ]
```

Reply:
Makes sense! Thanks